packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -368,14 +362,34 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
 
   //TODO: a better way to expose option parameter
   public startCall(participants: string[]): Call | undefined {
-    const idsToAdd = participants.map((participant) => {
-      // FIXME: `onStartCall` does not allow a Teams user.
-      // Need some way to return an error if a Teams user is provided.
-      const backendId = fromFlatCommunicationIdentifier(participant) as CommunicationUserIdentifier;
-      return backendId;
-    });
+    if (_isInCall(this.getState().call?.state ?? 'None')) {
+      throw new Error('You are already in the call.');
+    } else {
+      const idsToAdd = participants.map((participant) => {
+        // FIXME: `onStartCall` does not allow a Teams user.
+        // Need some way to return an error if a Teams user is provided.
+        const backendId = fromFlatCommunicationIdentifier(participant) as CommunicationUserIdentifier;
+        return backendId;
+      });
+
+      const call = this.handlers.onStartCall(idsToAdd);
```

Comment:
I'm a bit surprised that all these methods are synchronous. Doesn't starting a call take some time?
Same for the various operations inside `updateContextAfterJoiningCall`.

Reply:
I agree, callAgent.startCall and callAgent.join are the surprising ones to be synchronous -- but they are 🤷‍♀️ (most surprising is that is gets the call id for the call to join without being asynchronous)

Reply:
Pro-tip: `callId` can change during a call -- exactly for this reason. The client picks one, but the server can assign a different one once connected.